src/engine/sparqlExpressions/SparqlExpressionValueGetters.cpp

```diff
@@ -165,14 +165,68 @@ IntDoubleStr ToNumericValueGetter::operator()(
   AD_FAIL();
 }
 
-// ____________________________________________________________________________
+// _____________________________________________________________________________
 IntDoubleStr ToNumericValueGetter::operator()(
     const LiteralOrIri& s,
     [[maybe_unused]] const EvaluationContext* context) const {
   return std::string(asStringViewUnsafe(s.getContent()));
 }
 
-// ____________________________________________________________________________
+// _____________________________________________________________________________
+OptIri DatatypeValueGetter::operator()(ValueId id,
+                                       const EvaluationContext* context) const {
+  using enum Datatype;
+  auto datatype = id.getDatatype();
+  std::optional<std::string> entity;
+  switch (datatype) {
+    case Bool:
+      return Iri::fromIrirefWithoutBrackets(XSD_BOOLEAN_TYPE);
+    case Double:
+      return Iri::fromIrirefWithoutBrackets(XSD_DOUBLE_TYPE);
+    case Int:
+      return Iri::fromIrirefWithoutBrackets(XSD_INT_TYPE);
+    case Date: {
+      auto dateType = id.getDate().toStringAndType().second;
+      if (dateType != nullptr) {
```

Comment:
I looked at that function, it should never return `nullptr`.
So a simple `AD_CORRECTNESS_CHECK(dateType != nullptr)` should be sufficient and make the coverage tool happy.